src/main/java/apitests/steps/BaseSteps.java

```diff
@@ -0,0 +1,16 @@
+package apitests.steps;
+
+import apitests.clients.PetClient;
+import apitests.model.ResponseDetails;
+
+public class BaseSteps {
+
+    protected ResponseDetails response;
+
+    protected PetClient petClient;
+
+    public BaseSteps(ResponseDetails response, PetClient petClient) {
```

Comment:
Could you tell me where we pass those parameters to this class inside the test?

Reply:
It is done automatically by cucumber-picocontainer and dependency injection -> pass the state of pet object or response between the TestSteps classes if we use some common methods. Here you have a explanation of the problem I was traying to solve yesterday: https://grapeup.com/blog/dependency-injection-in-cucumber-jvm/

Reply:
I've tried to apply your suggestions but I loose again the response context between steps and test classes. In this shape it should stay at it is... I can try this approach without Cucumber and only with RestAssured itself. I hope that you are ok with that...